src/main/java/com/demo/jwt/JwtMybatisApplication/config/jwt/SecurityConfig.java

```diff
@@ -0,0 +1,73 @@
+package com.demo.jwt.JwtMybatisApplication.config.jwt;
+
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.jwk.OctetSequenceKey;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.oauth2.jose.jws.MacAlgorithm;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationConverter;
+import org.springframework.security.oauth2.server.resource.authentication.JwtGrantedAuthoritiesConverter;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+import javax.crypto.SecretKey;
+
+
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity(jsr250Enabled = true)
+public class SecurityConfig {
+
+    @Value("${jwt.secret}")
+    private String secretKey;
+
+    private static final String[] AUTH_WHITE_LIST = {
+            "/v3/api-docs/**",
+            "/swagger-ui/**",
+            "/v2/api-docs/**",
+            "/swagger-resources/**"
+    };
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity.authorizeHttpRequests(authorize ->
+                authorize
+                    .requestMatchers(AUTH_WHITE_LIST).permitAll()
+                    .anyRequest().authenticated()
+            )
+            .oauth2ResourceServer(configure -> configure.jwt(Customizer.withDefaults()))
+            .build();
+    }
+
+    @Bean
+    public JwtAuthenticationConverter jwtAuthenticationConverter() {
+        final JwtGrantedAuthoritiesConverter gac = new JwtGrantedAuthoritiesConverter();
+        gac.setAuthoritiesClaimName("Role");
+        gac.setAuthorityPrefix("ROLE_");
+        final JwtAuthenticationConverter jac = new JwtAuthenticationConverter();
+        jac.setJwtGrantedAuthoritiesConverter(gac);
+        return jac;
+    }
+
+    @Bean
+    public JwtDecoder jwtDecoder() {
+        return NimbusJwtDecoder.withSecretKey(stringToSecretKey(secretKey)).macAlgorithm(MacAlgorithm.HS512).build();
+    }
+
+    private SecretKey stringToSecretKey(String secretKey) {
```

Comment:
Please check if we need to specify the algorithm explicitly in both cases jwtDecoder and SecretKey.

Reply:
In case of HS256 algorithm don't need to specify algorithm to the decoder as authorization server uses HS256 as default algorithm but in case of HS512 we require to write(stackoverflow). I tried without explicitly providing HS512 algorithm to decoder but it was giving error like this: Signed JWT rejected: Another algorithm expected, or no matching key(s) found